### PR TITLE
cleanup azure leftovers

### PR DESCRIPTION
**Pull request by klensy**
Continuation of #97756

### CODE REVIEW

**Comment by klensy** [comment]
What part of https://github.com/rust-lang/rust/blob/f9097f87c9c094f80826fb60a1a624b5f9f1ed82/src/bootstrap/toolstate.rs#L358-L383
should be removed too?

can this be changed back to https?
https://github.com/rust-lang/rust/blob/f9097f87c9c094f80826fb60a1a624b5f9f1ed82/src/ci/run.sh#L153-L164

File: src/tools/miri/cargo-miri/src/util.rs [hunk 1]
```diff
@@ -169,10 +169,8 @@ where
 }
 
 pub fn ask_to_run(mut cmd: Command, ask: bool, text: &str) {
-    // Disable interactive prompts in CI (GitHub Actions, Travis, AppVeyor, etc).
-    // Azure doesn't set `CI` though (nothing to see here, just Microsoft being Microsoft),
-    // so we also check their `TF_BUILD`.
-    let is_ci = env::var_os("CI").is_some() || env::var_os("TF_BUILD").is_some();
```
Comment (taiki-e) [inline src/tools/miri/cargo-miri/src/util.rs:175]: I guess this will affect users who run Miri on Azure.

Comment (RalfJung) [reply src/tools/miri/cargo-miri/src/util.rs:175]: Yeah. Azure CI still exists, right? If so, `cargo miri` should recognize it.

**Comment by pietroalbini** [comment]
> What part of should be removed? 

Not really much, it's the same procedure even though it's now on GitHub Actions.

> can this be changed back to https?

I'd keep it as http: date drifts could cause validation of the tls certificate to fail, while the whole purpose of the check is to detect date drifts.

**Comment by pietroalbini** [comment]
Current changes look good, let me know if you want to make more here or if I should approve this :slightly_smiling_face:

**Comment by klensy** [comment]
> Current changes look good, let me know if you want to make more here or if I should approve this 🙂

Let's merge, no new ideas here yet.

**Pull request merged**